File: system-test/java/src/main/java/com/optivem/shop/testkit/dsl/port/then/ThenStage.java
package com.optivem.shop.testkit.dsl.port.then;

import com.optivem.shop.testkit.dsl.port.then.steps.ThenClock;
import com.optivem.shop.testkit.dsl.port.then.steps.ThenCountry;
import com.optivem.shop.testkit.dsl.port.then.steps.ThenProduct;

public interface ThenStage {
    ThenClock clock();

    ThenProduct product(String skuAlias);

    ThenCountry country(String countryAlias);
}

File: system-test/typescript/src/testkit/dsl/port/then/then-stage.ts
import type { ThenGivenClock } from './steps/then-given-clock.js';
import type { ThenGivenProduct } from './steps/then-given-product.js';
import type { ThenGivenCountry } from './steps/then-given-country.js';

export interface ThenStage {
  clock(): ThenGivenClock;
  product(skuAlias: string): ThenGivenProduct;
  country(countryAlias: string): ThenGivenCountry;
}
